compiler/circle-interpreter/src/CircleInterpreter_cffi.cpp

```diff
@@ -0,0 +1,210 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This library is for using luci-interpreter in Python. It is implemented
+ *  with CFFI. CFFI is a FFI (Foreign Function Interface) for Python calling
+ *  C code.
+ */
+
+#include <cstddef>
+#include <string>
+
+#include <luci/Importer.h>
+#include <luci_interpreter/Interpreter.h>
+
+namespace
+{
+
+// Global variable for error message
+std::string last_error_message;
+
+template <typename NodeT> size_t getTensorSize(const NodeT *node)
+{
+  uint32_t tensor_size = luci::size(node->dtype());
+  for (uint32_t i = 0; i < node->rank(); ++i)
+    tensor_size *= node->dim(i).value();
+  return tensor_size;
+}
+
+// Function to retrieve the last error message.
+extern "C" const char *get_last_error(void) { return last_error_message.c_str(); }
+
+// Clear the last error message
+extern "C" void clear_last_error(void) { last_error_message.clear(); }
+
+/**
+ * @brief A function that wraps another function and catches any exceptions.
+ *
+ * This function executes the given callable (`func`) with the provided arguments.
+ * If the callable throws an exception, the exception message is stored in a
+ *  `last_error_message` global variable.
+ *
+ * @tparam Func The type of the callable funciton.
+ * @tparam Args The types of arguments to pass to the callable function.
+ * @param func The callable function to execute.
+ * @param args The arguments to pass to the callable function.
+ * @return The return value of the callable function, or a default value in case of
+ *         an exception. If the function has a `void` return type, it simply returns
+ *         without any value.
+ *
+ * @note This function ensures that exceptions are safely caught and conveted to
+ *       error messages that can be queried externally, e.g. from Python.
+ */
+template <typename Func, typename... Args>
+auto exception_wrapper(Func func, Args... args) -> typename std::result_of<Func(Args...)>::type
+{
+  using ReturnType = typename std::result_of<Func(Args...)>::type;
+
+  try
+  {
+    return func(std::forward<Args>(args)...);
+  }
+  catch (const std::exception &e)
+  {
+    last_error_message = e.what();
+    if constexpr (not std::is_void<ReturnType>::value)
+    {
+      return ReturnType{};
+    }
+  }
+  catch (...)
+  {
+    last_error_message = "Unknown error";
+    if constexpr (not std::is_void<ReturnType>::value)
+    {
+      return ReturnType{};
+    }
+  }
+  // For void return type
+  if constexpr (std::is_void<ReturnType>::value)
+  {
+    return;
+  }
```

Comment:
This block gives no meaning as there is no further code after this block but just return.
Is this some python binding pattern? 

It may help adding some comments why we need to add this block.

Reply:
Oh, you're right. I'll remove this.